### ISSUE TITLE
[Enhancement] Upgrade pickers to Material M3 standards

### ISSUE THREAD

**Issue by gselderslaghs** [materializecss/materialize#558]
### Before submitting...

- [X] I have searched for [duplicate or closed issues](https://github.com/materializecss/materialize/issues?utf8=%E2%9C%93&q=is%3Aissue).
- [X] I have read the [CONTRIBUTING document](https://github.com/materializecss/materialize/blob/master/CONTRIBUTING.md) and my issue is following the guidelines.
- [X] I have read the template completely before filling it in.

### Context

Pickers could use an improvement as they currently do not comply with the Material M3 standards
This would require SCSS styling refinement on the components
Currently the pickers can only be rendered as modals, would be nice if additionally we can implement docked rendering options as requested per #140 

### Your Environment

* Version used: 2.1.1

**Comment by wuda-io** [comment]
Ok please wait with this change, i am refactoring it atm.

## For the modal part
I think we should remove the integrated Modal feom the pickers completely and let the user decide where the picker sits. Wdyt?

**Comment by gselderslaghs** [comment]
Agree with letting the user choose how the pickers should be rendered, having options "Modal - hovering in a popup" and "Docked - inline as a kind of dropdown below the field" available where only this latest options should present a new feature implementation. My idea with be with a "Docked" integration that we'd have the functionality the datepicker is visible when even the user is updating the main input field and represents the changes in real-time visually on the expanded picker

**Comment by gselderslaghs** [comment]
Additional enhancements (todo):
- [ ] Allow HTML time input field
- [ ] Implement functionality to allow timepicker to fill from element value in input field
- [ ] Allow format specification in option for rendering in input field

**Comment by gselderslaghs** [comment]
> The current state of the pickers is unusable. Is this still under development?
Yes, this currently only available in dev version and not in a stable release yet

> I try to replicate the previous behavior with adding 'container' to the options. But it is nowhere near how it is worked before this update. The callbacks works more or less with the datepicker but not with the timepicker. The OK and Close buttons are not working, got messages like "Datepicker.close() is deprecated. Remove this method and wrap in modal yourself."
Modal with confirmation buttons interaction buttons where deprecated and removed by in #525, there is currently no replacement in the stable release but some pieces are already in the v2-dev version:

Regarding the modal the idea is to implement a "docked" (Issue: #140, plugin: #582, timepicker implementation:#583, datepicker implementation: todo) and "modal" (todo) display plugin in the future

Confirmation buttons are available as utility (Timepicker #575, Datepicker #592) with optional confirmation buttons util (Discussion: #570) 

I would suggest to use the pull requests as patch in meanwhile for them to get confirmed by the maintainers of the project. I'm open to your feedback on the pull-requests as it would be easier for the maintainers as wel to validate

> Can you please update the documentation, how to correctly use the pickers with modal, that replicates the previous behavior.
Since it's still in dev release you'll have to pull and build the documentation by yourself, the available options in the dev release are available in the latest dev release of the documentation

**Comment by gselderslaghs** [comment]
Closing as fixed in v2-dev, please feel free to re-open if necessarily
